test/parallel/test-timers-unrefed-in-callback.js

```diff
@@ -0,0 +1,49 @@
+'use strict';
+// Check that unrefed setInterval timer keeps running even if it is
+// unrefed within callback.
+
+require('../common');
+const assert = require('assert');
+const net = require('net');
+
+let counter1 = 0;
+let counter2 = 0;
+
+// Test1 checks setInterval continues even if it is unrefed within
+// timer callback. counter1 continues to be incremented more than 11
+// until server close completed.
+function Test1() {
+  // server only for maintaining event loop
+  const server = net.createServer().listen(0);
+
+  const timer1 = setInterval(() => {
+    timer1.unref();
+    if (counter1++ === 10)
+      server.close(() => {
+        Test2();
+      });
+  }, 1);
+}
+
+
+// Test2 checks clearInterval also works fine for the unrefed timer
+// within timer callback. counter2 stops at exactly 11 with
+// clearInterval.
+function Test2() {
+  // server only for maintaining event loop
+  const server = net.createServer().listen(0);
+
+  const timer2 = setInterval(() => {
+    timer2.unref();
+    if (counter2++ === 10) {
+      clearInterval(timer2);
+      server.close();
+    }
+  }, 1);
+}
+
+process.on('exit', () => {
+  assert.strictEqual(counter2, 11);
```

Comment:
`counter2` increments up to 11 before `server.close()` so that the test is not failed even when server is closed immediately. The test can be finished without need to wait for the next event loop.